xds/type/matcher/v3/matcher.proto

```diff
@@ -125,18 +125,16 @@ message Matcher {
   }
 
   oneof matcher_type {
-    option (validate.required) = true;
-
     // A linear list of matchers to evaluate.
     MatcherList matcher_list = 1;
 
     // A match tree to evaluate.
     MatcherTree matcher_tree = 2;
   }
 
-  // Optional OnMatch to use if the matcher failed.
-  // If specified, the OnMatch is used, and the matcher is considered
-  // to have matched.
-  // If not specified, the matcher is considered not to have matched.
+  // Optional OnMatch to use if the matcher above was unspecified or if matching using the specified
```

Comment:
Suggest phrasing this comment as:

"""
Optional OnMatch to use if no matcher above matched (e.g., if there are no matchers specified above, or if none of the matches specified above succeeded).
If no matcher above matched and this field is not populated, the match will be considered unsuccessful.
"""